03_letters/nickolee/letter_frequency_test.go

```diff
@@ -0,0 +1,100 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"reflect"
+	"strconv"
+	"testing"
+)
+
+func TestMainOutput(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+
+	main()
+
+	expected := strconv.Quote(`a:2
+b:1
+`)
+	actual := strconv.Quote(buf.String())
+
+	if expected != actual {
+		t.Errorf("Unexpected output in main()\nexpected: %q\nactual: %q", expected, actual)
+	}
+}
+
+func TestLettersFunc(t *testing.T) {
+	type test struct {
+		name     string
+		input    string
+		expected map[rune]int
+	}
+
+	tests := []test{
+		{name: "empty string", input: "", expected: map[rune]int{}}, // expect empty map
+		{name: "ASCII string", input: "  ``~8I,dzye[uY6<mCh<n 9Otefp0fX0-@2<C)z)}.go-Hq{n]LX 8uKnRxj (92@08b9P ",
+			expected: map[rune]int{
```

Comment:
This must have taken a while to type up.

Reply:
A terrible mouthful...